phpcs/Standards/QualityAssurance/Sniffs/Generic/HardcodedPathSniff.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * QualityAssurance_Sniffs_Generic_HardcodedPathSniff.
+ *
+ * PHP version 5
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     http://pear.php.net/package/PHP_CodeSniffer
+ */
+
+/**
+ * Hardcoded paths to theme or modules need to be removed or replace by a call
+ * to drupal_get_path() function.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     http://pear.php.net/package/PHP_CodeSniffer
+ */
+class QualityAssurance_Sniffs_Generic_HardcodedPathSniff implements PHP_CodeSniffer_Sniff
+{
+  /**
+   * Returns an array of tokens this test wants to listen for.
+   *
+   * @return array
+   */
+  public function register()
+  {
+    return array(
+      T_CONSTANT_ENCAPSED_STRING,
```

Comment:
Think T_CONSTANT_ENCAPSED_STRING only was sufficient. But we have to test all token types. There was actually no reason for me to grab so many tokens in the other sniffs... But as usual I find it hard to set aside time for code that's functional but not optimal... 😊